lib/Map/ImageryProvider/ProtomapsImageryProvider.ts

```diff
@@ -248,69 +230,91 @@ export default class ProtomapsImageryProvider
     return [];
   }
 
-  async requestImage(x: number, y: number, level: number) {
+  async requestImage(x: number, y: number, level: number, request?: Request) {
     const canvas = document.createElement("canvas");
     canvas.width = this.tileWidth;
     canvas.height = this.tileHeight;
-    return await this.requestImageForCanvas(x, y, level, canvas);
+    return await this.requestImageForCanvas(x, y, level, canvas, request);
   }
 
   async requestImageForCanvas(
     x: number,
     y: number,
     level: number,
-    canvas: HTMLCanvasElement
+    canvas: HTMLCanvasElement,
+    request?: Request
   ) {
-    try {
-      await this.renderTile({ x, y, z: level }, canvas);
-    } catch (e) {
-      console.log(e);
-    }
+    if (isDefined(this.softMinimumLevel) && level < this.softMinimumLevel)
+      throw TerriaError.from(
+        `Level: ${level} is below softMinimumLevel (=${this.softMinimumLevel})`
+      );
 
-    return canvas;
-  }
+    const coords: Coords = { z: level, x, y };
 
-  public async renderTile(coords: Coords, canvas: HTMLCanvasElement) {
     // Adapted from https://github.com/protomaps/protomaps.js/blob/master/src/frontends/leaflet.ts
-    let tile: PreparedTile | undefined = undefined;
+    let tile: PreparedTile;
 
     // Get PreparedTile from source or view
     // Here we need a little bit of extra logic for the ProtomapsGeojsonSource
-    if (this.source instanceof ProtomapsGeojsonSource) {
-      const data = await this.source.get(coords, this.tileHeight);
+    if (
+      this.source instanceof ProtomapsGeojsonSource ||
+      this.source instanceof ProtomapsArcGisPbfSource
+    ) {
+      const data = await this.source.get(coords, this.tileHeight, request);
 
       tile = {
         data: data,
         z: coords.z,
         dataTile: coords,
         scale: 1,
-        origin: new Point(coords.x * 256, coords.y * 256),
+        origin: new Point(
+          coords.x * PROTOMAPS_DEFAULT_TILE_SIZE,
+          coords.y * PROTOMAPS_DEFAULT_TILE_SIZE
+        ),
         dim: this.tileWidth
       };
     } else if (this.view) {
       tile = await this.view.getDisplayTile(coords);
+    } else {
+      throw TerriaError.from(
+        `Failed to get tile - no view or appropriate source in ProtomapsImageryProvider`
+      );
     }
 
-    if (!tile) return;
-
     const tileMap = new Map<string, PreparedTile[]>().set("", [tile]);
 
     this.labelers.add(coords.z, tileMap);
 
     const labelData = this.labelers.getIndex(tile.z);
 
     const bbox = {
-      minX: 256 * coords.x - PROTOMAPS_TILE_BUFFER,
-      minY: 256 * coords.y - PROTOMAPS_TILE_BUFFER,
-      maxX: 256 * (coords.x + 1) + PROTOMAPS_TILE_BUFFER,
-      maxY: 256 * (coords.y + 1) + PROTOMAPS_TILE_BUFFER
+      minX: PROTOMAPS_DEFAULT_TILE_SIZE * coords.x - PROTOMAPS_TILE_BUFFER,
```

Comment:
Is the RHS here `origin.x - PROTOMAPS_TILE_BUFFER`?

Reply:
not sure why you think so.